password_security/controllers/main.py

```diff
@@ -70,17 +70,32 @@ def web_auth_reset_password(self, *args, **kw):
             request.httprequest.method == "POST"
             and qcontext.get("login")
             and "error" not in qcontext
-            and "token" not in qcontext
+            and not qcontext.get("token")
         ):
             login = qcontext.get("login")
-            user_ids = request.env.sudo().search(
-                [("login", "=", login)],
-                limit=1,
-            )
-            if not user_ids:
-                user_ids = request.env.sudo().search(
-                    [("email", "=", login)],
+            user = (
+                request.env["res.users"]
+                .sudo()
+                .search(
+                    [("login", "=", login)],
                     limit=1,
                 )
-            user_ids._validate_pass_reset()
-        return super(PasswordSecurityHome, self).web_auth_reset_password(*args, **kw)
+            )
+            if not user:
+                user = (
+                    request.env["res.users"]
+                    .sudo()
+                    .search(
+                        [("email", "=", login)],
+                        limit=1,
+                    )
+                )
+            try:
+                user._validate_pass_reset()
+            except UserError as e:
+                qcontext["error"] = e.args[0]
+                response = request.render("auth_signup.reset_password", qcontext)
+                response.headers["X-Frame-Options"] = "DENY"
+                return response
```

Comment:
What if the call to _validate_pass_reset moved into `res.users` action_reset_password instead? That would remove the need for this copy pasta.

Reply:
It might be necessary to check to see if the `request.uid` is the admin user, to allow for bulk password resets that ignore this setting.

Reply:
I commit https://github.com/OCA/server-auth/pull/492/commits/8a5ead9447324199017fef6b984ba17302bcc2e6 I moved `_validate_pass_reset` into `action_reset_password`. It seems working fine to me. Would you also give it a look?